engine/api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentProvider.scala

```diff
@@ -0,0 +1,116 @@
+package pl.touk.nussknacker.engine.api.component
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.api.process.{ProcessObjectDependencies, SingleNodeConfig}
+
+
+/*
+  Service, SourceFactory, SinkFactory, CustomStreamTransformer
+ */
+trait Component
+
+
+/*
+  componentProviders {
+    sqlHsql {
+      provider: sql
+      jdbcUrl: jdbc:hsql...//
+      admin/pass
+    }
+    sqlOracle {
+      provider: sql
+      jdbcUrl: jdbc:oracle...//
+      admin/pass
+    }
+    prinzH20 {
+      provider: prinzH20
+      h2oLocation:
+    }
+    prinzMlFlow {
+      //this is not needed, as configuration is named just as provider
+      //provider: prinzMLFlow
+      mlFlowUrl:
+    }
+    openapiService1 {
+      provider: openapi
+      url: http://host1
+    }
+    openapiService2 {
+      provider: openapi
+      url: http://host2
+      componentsConfig {
+        //we want to ignore particular service here
+        serviceA {
+          disabled: true
+        }
+      }
+    }
+    //not needed in our domain
+    aggregation {
+      disabled: false
+    }
+    configurableFilter {
+      componentsConfig: {
+        filterWithA {
+          component: customFilter
+          customProperty1: A
+        },
+        filterWithB {
+          component: customFilter
+          customProperty1: B
+        }
+      }
+    }
+  }
+ */
+case class ComponentProviderConfig(provider: Option[String],
```

Comment:
What is `provider` here? Shouldn't it be as a key in map of provider configs? Why it is optional?

Reply:
My idea was to make easy things simple and compex thigns possible:
`
    openapiService1 {
      provider: openapi //we have different configurations of openapi, each is named
      url: http://host1
    }
    singleService {
      //provider: singleService //if have only one configuration of provider, name is taken from name of configuration
    }
`
I think in many cases `singleService` way will be ok

Reply:
ok, let's name it `providerType`

Reply:
:+1: